### PR TITLE
feat:Created doctype Employee Exit Clearance and child table Employee Clearance.

### PR DESCRIPTION
**Pull request by ZakkiyathLulu**
## Feature description
Create  doctype  Employee Exit Clearance and child table Employee Clearance.

## Analysis and design (optional)
Create  doctype  Employee Exit Clearance with following fields:

- Role Permission-HR Manager,HOD
- It should be Submittable doctype
- Employee(Link,option-Employee,mandatory)
- Employee Name(Read-only,fetch from employee)
- Department(link to Department,fetch from employee)
- Designation(link to designation,fetch from employee)
- Employee Separation Begins on(Date)
- Relieving Date(Date,fetch from employee)
- Notice Period(Int,fetch from employee)
- Pending Dues(check)
- Amount(Currency,show this field only if Pending Dues is checked)

Create childtable  Employee Clearance :
-Department(link,option-department)
- Employee Exit Clearance(link,option-Employee Exit Clearance)
- Status(select,options-pending,completed)-----set status to completed when the Employee Exit Clearance is submitted

## Solution description
Created doctype  Employee Exit Clearance with above mentioned fields and permissions set to HOD and HR Manager.

## Output screenshots (optional)

image
image

## Areas affected and ensured
Employee Exit  Clearance
Employee  Clearance 

## Is there any existing behavior change of other features due to this code change?
 No. 

## Was this feature tested on the browsers?
  - Chrome
  - Mozilla Firefox

### CODE REVIEW

File: beams/beams/doctype/employee_clearance/employee_clearance.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   "label": "Employee Exit Clearance",
+   "options": "Employee Exit Clearance"
+  },
+  {
+   "fieldname": "status",
+   "fieldtype": "Select",
+   "in_list_view": 1,
+   "label": "Status",
+   "options": "\nPending\nCompleted"
+  }
```
Comment (SherinKR) [inline beams/beams/doctype/employee_clearance/employee_clearance.json:34]: Bring status to right. Is \n required in option

File: beams/beams/doctype/employee_exit_clearance/employee_exit_clearance.json [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+   "fieldname": "employee_name",
+   "fieldtype": "Data",
+   "label": "Employee Name"
+  },
+  {
+   "fetch_from": "employee.department",
+   "fieldname": "department",
+   "fieldtype": "Data",
+   "label": "Department"
+  },
```
Comment (SherinKR) [inline beams/beams/doctype/employee_exit_clearance/employee_exit_clearance.json:45]: Department should be Link

File: beams/beams/doctype/employee_exit_clearance/employee_exit_clearance.json [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  {
+   "fetch_from": "employee.designation",
+   "fieldname": "designation",
+   "fieldtype": "Data",
+   "label": "Designation"
+  },
```
Comment (SherinKR) [inline beams/beams/doctype/employee_exit_clearance/employee_exit_clearance.json:51]: Designation should be Link

File: beams/beams/doctype/employee_exit_clearance/employee_exit_clearance.json [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+   "fieldname": "pending_dues",
+   "fieldtype": "Check",
+   "label": "Pending Dues"
+  },
+  {
+   "depends_on": "eval:doc.pending_dues;",
+   "fieldname": "amount",
+   "fieldtype": "Float",
+   "label": "Amount"
+  },
```
Comment (SherinKR) [inline beams/beams/doctype/employee_exit_clearance/employee_exit_clearance.json:74]: It should be Currency

**Review by SherinKR — CHANGES_REQUESTED** [review]

File: beams/beams/doctype/employee_exit_clearance/employee_exit_clearance.json [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  {
+   "fieldname": "employee_separation_begins_on",
+   "fieldtype": "Date",
+   "label": "Employee Separation Begins On"
+  },
+  {
+   "fetch_from": "employee.relieving_date",
+   "fieldname": "relieving_date",
+   "fieldtype": "Data",
```
Comment (SherinKR) [inline beams/beams/doctype/employee_exit_clearance/employee_exit_clearance.json:62]: Change to Date

**Review by SherinKR — CHANGES_REQUESTED** [review]

**Review by SherinKR — APPROVED** [review]

**Review by MhmdSinanKT — APPROVED** [review]

**Pull request merged**